packages/pagination-hoc/src/pagination.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+import React, { Component } from 'react';
+
+import type { Options, Props, State } from './types';
+import type { ComponentType } from 'react';
+
+const withPagination = (options: Options) => (WrapperComponent: ComponentType<Props>) => {
+  return class extends Component<Props, State> {
+    state = {
+      page: +new URLSearchParams(window.location.search).get('page') || 1,
+    };
+
+    pushNewLocation = (params: any) => this.props.history.push(window.location.pathname + '?' + params.toString());
```

Comment:
`this.props.location` should be used here.

If we remove history.push from this HOC it will be useless. Cause auto adding/changing page query in window.location is the only job that this HOC handle right now.